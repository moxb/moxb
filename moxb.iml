<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>