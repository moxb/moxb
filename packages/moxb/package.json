{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
