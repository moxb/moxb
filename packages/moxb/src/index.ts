export * from './impl';
export * from './types';
export * from './routing';
