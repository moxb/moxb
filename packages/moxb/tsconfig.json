{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
