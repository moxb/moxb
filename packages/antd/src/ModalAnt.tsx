import { Modal as MoxbModal, ModalActions } from '@moxb/moxb';
import { ModalFuncProps } from 'antd/lib/modal';
import { observer } from 'mobx-react';
import * as React from 'react';
import { Modal } from 'antd';

export interface BindModalAntProps<T, A extends ModalActions = ModalActions> extends ModalFuncProps {
    operation: MoxbModal<T, A>;
    footer?(actions: A): React.ReactNode;
}

// same sizes as semantic UI's modals
const SIZES = {
    default: '900px', // default size for ant modal: 512px
    mini: '360px',
    tiny: '540px',
    small: '720px',
    large: '1080px',
    fullscreen: '95% !important',
};

@observer
export class ModalAnt<T, A extends ModalActions = ModalActions> extends React.Component<BindModalAntProps<T, A>> {
    /*
     * Ant Design has no concept for multiple actions in a modal dialog, it will always be an 'okButton' and
     * a 'cancelButton', so we will warn the user, that not more actions are available.
     * */
    render() {
        const { operation, children, footer, width, ...props } = this.props;
        const size = this.props.width || SIZES[operation.size || 'default'];

        if (footer && operation.actions) {
            return (
                <Modal
                    {...props as any}
                    visible={operation.open}
                    title={operation.header}
                    onCancel={operation.actions.cancel.fire}
                    footer={footer(operation.actions)}
                    width={size}
                >
                    {children}
                </Modal>
            );
        }

        return (
            <Modal
                {...props as any}
                visible={operation.open}
                onCancel={operation.actions.cancel.fire}
                cancelText={operation.actions.cancel.label}
                okText={operation.actions.confirm && operation.actions.confirm.label}
                onOk={operation.actions.confirm && operation.actions.confirm.fire}
                title={operation.header}
                width={size}
            >
                {children}
            </Modal>
        );
    }
}
