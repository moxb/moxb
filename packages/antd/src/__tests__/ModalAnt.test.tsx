import { ActionImpl, ModalImpl, ModalActions } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { ModalAnt } from '../ModalAnt';
import { shallowMoxbToJson } from './enzymeHelper';

interface CustomModalActions extends ModalActions {
    test1: string;
    test2: string;
}

describe('ModalAnt', function() {
    it('should render a modal dialog by default', function() {
        const operation = new ModalImpl<any, ModalActions>({
            actions: () => ({
                cancel: new ActionImpl({
                    id: 'Id.btn',
                    label: 'Action',
                    fire: jest.fn(),
                }),
                confirm: new ActionImpl({
                    id: 'Id.btn1',
                    label: 'Cancel',
                    fire: jest.fn(),
                }),
            }),
            header: () => 'New Modal Header',
        });
        expect(shallowMoxbToJson(shallow(<ModalAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should support a custom footer', function() {
        const operation = new ModalImpl<any, CustomModalActions>({
            actions: () => ({
                cancel: new ActionImpl({
                    id: 'Id.btn',
                    label: 'Action',
                    fire: jest.fn(),
                }),
                test1: 'Hello',
                test2: 'World',
            }),
            header: () => 'New Modal Header',
        });
        expect(
            shallowMoxbToJson(
                shallow(
                    <ModalAnt
                        operation={operation}
                        footer={actions => <div>{`${actions.test1} ${actions.test2}`}</div>}
                    />
                )
            )
        ).toMatchSnapshot();
    });
});
