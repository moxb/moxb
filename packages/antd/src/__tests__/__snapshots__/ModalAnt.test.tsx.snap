// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAnt should render a modal dialog by default 1`] = `
<Modal
  cancelButtonDisabled={false}
  cancelText="Action"
  confirmLoading={false}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okText="Cancel"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title=""
  transitionName="zoom"
  visible={false}
  width="900px"
/>
`;

exports[`ModalAnt should support a custom footer 1`] = `
<Modal
  cancelButtonDisabled={false}
  confirmLoading={false}
  footer={
    <div>
      Hello World
    </div>
  }
  maskTransitionName="fade"
  okButtonDisabled={false}
  okType="primary"
  onCancel={[Function]}
  title=""
  transitionName="zoom"
  visible={false}
  width="900px"
/>
`;
